### PR TITLE
Added unit and integration testing for spire to chains

### PR DESCRIPTION
**Pull request by pxp928**

### CODE REVIEW

File: pkg/config/store_test.go [hunk 1]
```diff
@@ -150,6 +154,10 @@ func TestParse(t *testing.T) {
 				Transparency: TransparencyConfig{
 					URL: "https://rekor.sigstore.dev",
 				},
+				SPIRE: SPIREConfig{
+					Enabled:    false,
+					SocketPath: "/spiffe-workload-api/spire-agent.sock",
```
Comment (lumjjb) [inline pkg/config/store_test.go:160]: nit: if enabled is false let's remove this, if not it may hint that the path is still somehow necessary

Comment (lumjjb) [reply pkg/config/store_test.go:160]: on a side note, can we do one enabled with spireMock flag like we did in pipelines? https://github.com/tektoncd/pipeline/pull/4759/files#diff-c199494aad61a99b439167d951412a7ca76efb20d52715d95e6796ae437a335eR33-R36

File: test/e2e-common.sh [hunk 2]
```diff
@@ -59,6 +59,101 @@ function install_pipeline_crd() {
   wait_until_pods_running tekton-pipelines || fail_test "Tekton Pipeline did not come up"
 }
 
+function spire_apply() {
```
Comment (lumjjb) [inline test/e2e-common.sh:65]: nit: add a comment here describing function and arguments

**Review by lumjjb — APPROVED** [review]
minor nits, lgtm

File: test/e2e-common.sh (hunk 2 above)
Comment (pxp928) [reply test/e2e-common.sh:65]: added.

File: pkg/config/store_test.go (hunk 1 above)
Comment (pxp928) [reply pkg/config/store_test.go:160]: Added the enabled flag. Didnt need to mock it out.

Comment (pxp928) [reply pkg/config/store_test.go:160]: Using the mocked spire client for the format_test.go
```
	spireMockClient := &spire.MockClient{}
	var (
		cc spire.ControllerAPIClient = spireMockClient
	)
```

**Pull request merged**